### PR TITLE
Implement kpod rm

### PR DESCRIPTION
**Pull request by rhatdan**
Kpod rm removes a container from the system

Signed-off-by: Ryan Cole <rcyoalne@gmail.com>

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
This seems to be missing the changes in libkpod to enable rm?

**Comment by mheon** [comment]
@rhatdan I think you're missing a commit or two

**Comment by rhatdan** [comment]
@mheon Yup.  Added them back into the pull request.

**Comment by rhatdan** [comment]
[retest]

File: libkpod/remove.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	ctrID := ctr.ID()
+
+	if err := c.runtime.UpdateStatus(ctr); err != nil {
+		return "", errors.Wrapf(err, "failed to update container state")
+	}
+
+	err = c.runtime.UpdateStatus(ctr)
+	if err != nil {
+		return "", errors.Wrapf(err, "could not update status for container %s", ctrID)
+	}
```
Comment (nalind) [inline libkpod/remove.go]: Are we supposed to be calling `UpdateStatus` twice?

File: cmd/kpod/rm.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		Action:      rmCmd,
+		ArgsUsage:   "",
+	}
+)
+
+// saveCmd saves the image to either docker-archive or oci
+func rmCmd(c *cli.Context) error {
+	args := c.Args()
+	if len(args) == 0 {
+		return errors.Errorf("need at least 1 argument")
```
Comment (mheon) [inline cmd/kpod/rm.go]: "need at least 1 container to remove"?

File: libkpod/remove.go (hunk 1 above)
Comment (mheon) [inline libkpod/remove.go]: I think the second one is duplication and can safely be removed.

File: libkpod/remove.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+// ContainerRemove removes a r
+func (c *ContainerServer) ContainerRemove(container string, force bool) (string, error) {
+	ctr, err := c.LookupContainer(container)
+	if err != nil {
+		return "", err
+	}
+	ctrID := ctr.ID()
+
+	if err := c.runtime.UpdateStatus(ctr); err != nil {
+		return "", errors.Wrapf(err, "failed to update container state")
```
Comment (mheon) [inline libkpod/remove.go]: Actually, we can remove this entirely

File: libkpod/remove.go (hunk 1 above)
Comment (mheon) [inline libkpod/remove.go]: On further investigation, it looks like the changes @mrunalp made earlier removed any call to update in ContainerRemove, so we can remove both of these calls safely

File: libkpod/stop.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+// ContainerStop stops a running container with a grace period (i.e., timeout).
+func (c *ContainerServer) ContainerStop(container string, timeout int64) (string, error) {
+	ctr, err := c.LookupContainer(container)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to find container %s", container)
+	}
+
+	if err := c.runtime.UpdateStatus(ctr); err != nil {
```
Comment (mheon) [inline libkpod/stop.go]: Remove this UpdateStatus call as well

File: server/container_remove.go [hunk 5]
```diff
@@ -14,42 +9,11 @@ import (
...
-		if err := s.StorageRuntimeServer().StopContainer(c.ID()); err != nil {
-			return nil, fmt.Errorf("failed to unmount container %s: %v", c.ID(), err)
-		}
-	}
-
-	if err := s.Runtime().DeleteContainer(c); err != nil {
-		return nil, fmt.Errorf("failed to delete container %s: %v", c.ID(), err)
-	}
-
-	if err := os.Remove(filepath.Join(s.config.ContainerExitsDir, c.ID())); err != nil {
```
Comment (mheon) [inline server/container_remove.go]: This is missing from the new code in libkpod

File: libkpod/remove.go (hunk 1 above)
Comment (rhatdan) [inline libkpod/remove.go]: Ok Removed.

File: cmd/kpod/rm.go (hunk 2 above)
Comment (rhatdan) [inline cmd/kpod/rm.go]: Fixed

File: libkpod/remove.go (hunk 3 above)
Comment (rhatdan) [inline libkpod/remove.go]: gone.

File: libkpod/stop.go (hunk 4 above)
Comment (rhatdan) [inline libkpod/stop.go]: Removed.

File: server/container_remove.go (hunk 5 above)
Comment (rhatdan) [inline server/container_remove.go]: Added back, nice catch.

**Pull request closed without merging**